apps/server/src/api/v2/file-upload.ts

```diff
@@ -14,3 +21,17 @@ function handleFileUpload(ctx) {
     },
   };
 }
+
+function delPrevFilesIfAny(currentFileName) {
+  const uploadsDir = config.uploadsPath;
+  return getFileNames(uploadsDir).then(files => {
+    const resourceId = currentFileName.split('-')[0];
+    const filesToDel = files.filter(
+      file => file.substr(0, file.indexOf('-')) === resourceId && file !== currentFileName
+    );
+    filesToDel.map(async fileToDel => {
+      const filePathToDel = path.join(uploadsDir, fileToDel);
+      await promisify(fs.unlink)(filePathToDel);
```

Comment:
we can promisify fs.unlink at the top of the file to avoid promisifying it everytime in this loop